slick/src/main/scala/slick/jdbc/JdbcDataSource.scala

```diff
@@ -99,7 +99,9 @@ object DataSourceJdbcDataSource extends JdbcDataSourceFactory {
         BeanConfigurator.configure(ds, c.toProperties, Set("url", "user", "password", "properties", "driver", "driverClassName"))
         ds
     }
-    new DataSourceJdbcDataSource(ds, c.getBooleanOr("keepAliveConnection"), None, new ConnectionPreparer(c))
+    val numThreads = c.getIntOr("numThreads", 20)
+    val maxConnections = Some(c.getIntOr("maxConnections", numThreads * 5))
```

Comment:
This is incorrect in the case where `dataSourceClass` is `None`: `maxConnections` should be too. Could move into the match statement above `val (ds, maxConnections) = ...`

Reply:
Yes, but in any case we should also not try to come up with a value for `maxConnections` like I tried. 

When we do have a `dataSourceClass` defined, the `maxConnections` should be:
`val maxConnections = c.getIntOpt("maxConnections")`